### PR TITLE
feat(top-app-bar): Add top app bar fixed variant

### LINKED ISSUE #54: Add Top App Bar's new Fixed Variant
After upgrading MDC Web to v0.35.1, the fixed variant must be added to the React Component.

### PR DESCRIPTION
**Pull request by moog16**
fixes #54
Related to changes from v0.35.x web sync

### CODE REVIEW

File: packages/top-app-bar/index.js [hunk 1]
```diff
@@ -67,6 +77,17 @@ export default class TopAppBar extends React.Component {
...
 
+  setStyle = (varName, value) => {
+    const updatedStyle = Object.assign({}, this.state.style);
+    updatedStyle[varName] = value;
+    this.setState({style: updatedStyle});
+  }
+
+  getMergedStyles = () => {
+    const {style} = this.state;
+    return Object.assign({}, style, this.props.style);
```
Comment (bonniezhou) [inline packages/top-app-bar/index.js]: Is there a reason we don't want to add `props.style` to the `state`? What if we add `props.style` to the state in the constructor, and just return `state.style` here?

File: .travis.yml [hunk 2]
```diff
@@ -31,5 +31,5 @@ matrix:
       install: npm i
       script:
         - ./test/screenshot/start.sh
-        - sleep 190s
+        - sleep 200s
```
[the remark this answers is not in the document]
Comment (moog16) [reply .travis.yml]: Every time a new screenshot test is added I need to increase the time. Its a guess and check time increase, but its been averaging about 10 - 20 ms per new test.

File: test/screenshot/top-app-bar/prominent.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ import './index.scss';
 import MaterialIcon from '../../../packages/material-icon';
 
 ReactDOM.render((
-  <div>
+  <div className='top-app-bar-container'>
```
[the remark this answers is not in the document]
Comment (moog16) [reply test/screenshot/top-app-bar/prominent.js]: oh ya - I would like to get Stackblitz or something like it working here. Created this card https://github.com/material-components/material-components-web-react/issues/84

File: packages/top-app-bar/index.js [hunk 4]
```diff
@@ -67,6 +77,17 @@ export default class TopAppBar extends React.Component {
     return React.cloneElement(element, {hasRipple: true});
   }
 
+  setStyle = (varName, value) => {
```
[the remark this answers is not in the document]
Comment (moog16) [reply packages/top-app-bar/index.js]: I thought it cleaned up the adapter, but no it won't be used anywhere else. Will move it

File: packages/top-app-bar/index.js (hunk 1 above)
Comment (moog16) [reply packages/top-app-bar/index.js]: I thought this was the simplest approach. We update this.state.style in the setStyle adapter method. So every time we call setStyle, we would need to merge this.props.style. Also if this.props.style every updates, we would need to merge it again with this.state.style.

I think doing it in this method is the most straightforward.

Comment (bonniezhou) [reply packages/top-app-bar/index.js]: Oh I see, this makes sense if we don't want to have to keep track of whether or not props.style was merged into the state.

File: test/unit/top-app-bar/index.test.js [hunk 5]
```diff
@@ -162,6 +173,34 @@ test('#adapter.getTotalActionItems returns false with no actionItem ' +
...
+
+test('#adapter.getTopAppBarHeight should return clientHeight', () => {
+  const div = document.createElement('div');
+  // needs to be attached to real DOM to get width
+  // https://github.com/airbnb/enzyme/issues/1525
+  document.body.append(div);
+  const options = {attachTo: div};
+  const wrapper = mount(<TopAppBar title='Hi' />, options);
+  const topAppBarHeight = wrapper.instance().adapter.getTopAppBarHeight();
+  assert.equal(topAppBarHeight, 18);
```
Comment (bonniezhou) [inline test/unit/top-app-bar/index.test.js]: Nit: Add a comment about where 18 comes from

**Pull request merged**